File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/core/BasePresenter.java
package com.przemyslawlusnia.vocabularycreator.core;

public abstract class BasePresenter<V> {

  public V view;

  public void onAttachView(V frag) {
    view = frag;
  }

  public void onDestroy() {
    view = null;
  }

  public abstract void onUnsubscribe();

}

File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/core/BasePresenter.kt
package com.przemyslawlusnia.vocabularycreator.core

abstract class BasePresenter<V> {

    protected var view: V? = null

    fun onAttachView(frag: V) {
        view = frag
    }

    fun onDetach() {
        view = null
    }

    abstract fun onUnsubscribe()

}
